lib/util/http/schema.ts

```diff
@@ -0,0 +1,28 @@
+import { ZodError, ZodIssue } from 'zod';
+import { SCHEMA_ERROR } from '../../constants/error-messages';
+
+export interface Schema<Output> {
+  parse(input: unknown): Output;
+}
+
+export interface HttpSchema<Req, Resp> {
+  request?: Schema<Req>;
+  response?: Schema<Resp>;
+}
```

Comment:
@rarkins @viceice What do you think about new schema parameter for requests? It will be used mostly for `response` part, however we may want validate our request data too.